src/Language/Printer.php

```diff
@@ -2,17 +2,55 @@
 
 namespace Digia\GraphQL\Language;
 
+use Digia\GraphQL\Error\LanguageException;
 use Digia\GraphQL\Language\AST\Node\NodeInterface;
+use Digia\GraphQL\Language\Writer\WriterInterface;
 
 class Printer implements PrinterInterface
 {
+    /**
+     * @var array|WriterInterface[]
+     */
+    protected $writers;
+
+    /**
+     * Printer constructor.
+     * @param array|WriterInterface[] $writers
+     */
+    public function __construct(array $writers)
+    {
+        foreach ($writers as $writer) {
+            $writer->setPrinter($this);
+        }
+
+        $this->writers = $writers;
+    }
 
     /**
      * @inheritdoc
+     * @throws LanguageException
      */
     public function print(NodeInterface $node): string
```

Comment:
I guess NodeInterface is still a blank interface?

Reply:
No, there are some common methods.